chingu-voyages/v40-geckos-team-20
# Feature/consistent button style - EC-122

@hmshp please review this PR
@Paulette-Zaldivar-Flores FYI - feel free to review too if you like

This PR covers ticket: https://viteshbava.atlassian.net/browse/EC-122 - introducing a button component so styles for buttons can be managed in one place.

To test:
* Just ensure all the button are still behaving/rendering as expected: filter drop downs, search, back cta.

Background / changes (in case you're interested - no need to read if you don't want to):
* This PR resolves the small "bug" I introduced with the Back button in the header where in adding the clickable link behind the logo, I had to change the layout properties of the header to make the elements behave correctly, but this ended up making the link only clickable on the text and not on the entire red part of the button.
* I also used this as an opportunity to create a global button component that can be re-used across the site (probably not necessary in this small website but it's evidence of a good practice I guess).  The component provides either a styled component, or just the css that can be used in other components.
* Hyemi, in the header you had a BackCTAWrapper component which I think you used so you could position/style the link button correctly.  With this new component, I think we don't need that anymore, and we can just make use of the button css, and apply more specific css through a class (which is what I have implemented).  Also takes out the need for an extra div in the html that isn't needed.
